pageserver/src/tenant/mgr.rs

```diff
@@ -453,16 +542,19 @@ pub async fn create_tenant(
     ctx: &RequestContext,
 ) -> Result<Arc<Tenant>, TenantMapInsertError> {
     tenant_map_insert(tenant_id, || async {
+
+        let location_conf = LocationConf::new(tenant_conf, generation);
+
         // We're holding the tenants lock in write mode while doing local IO.
         // If this section ever becomes contentious, introduce a new `TenantState::Creating`
         // and do the work in that state.
-        let tenant_directory = super::create_tenant_files(conf, tenant_conf, &tenant_id, CreateTenantFilesMode::Create).await?;
+        let tenant_directory = super::create_tenant_files(conf, location_conf.clone(), &tenant_id, CreateTenantFilesMode::Create).await?;
```

Comment:
Let's make this `&location_conf`, there's no reason why `persist_tenant_config` needs an owned copy of the location_conf.

Reply:
Yep, this was just an existing path that was using by-value TenantConfOpt for some reason.  Fixed in 69e80d012